From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Sat, 2 Dec 2023 22:53:02 +0000
Subject: [PATCH 36/67] Add and adjust embedlite static prefs

Adds the various embedlite static preferences to StaticPrefList.yaml so
that these will be availble for use as static pref variables in the C++
code.

[sailfishos][gecko] Set security.external_protocol_requires_permission to default to false

Upstream introduces the new user preference:

security.external_protocol_requires_permission

This defaults to true, but we need it to default to false so that
external URL actions continue to work as intended. Here we change the
default value of the preference from true to false so as to allow this.

---
 modules/libpref/init/StaticPrefList.yaml      | 23 ++++++++
 modules/libpref/init/all.js                   |  2 +-
 modules/libpref/moz.build                     |  1 +
 .../handling/ContentDispatchChooser.sys.mjs   |  2 +-
 4 files changed, 26 insertions(+), 2 deletions(-)

diff --git a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
index eeda4cc25412..be2cbb18025f 100644
--- a/modules/libpref/init/StaticPrefList.yaml
+++ b/modules/libpref/init/StaticPrefList.yaml
@@ -5075,6 +5075,29 @@
 #endif
   mirror: always
 
+#ifdef MOZ_EMBEDLITE
+  #---------------------------------------------------------------------------
+  # Prefs starting with "embedlite."
+  #---------------------------------------------------------------------------
+
+# Default azpc notifications behavior
+-   name: embedlite.azpc.handle.viewport
+    type: bool
+    value: true
+    mirror: always
+
+-   name: embedlite.azpc.handle.singletap
+    type: bool
+    value: true
+    mirror: always
+
+-   name: embedlite.azpc.handle.longtap
+    type: bool
+    value: true
+    mirror: always
+
+#endif
+
 #---------------------------------------------------------------------------
 # Prefs starting with "extensions."
 #---------------------------------------------------------------------------
diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index 3f860aa32061..ae7ddd934dc7 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -4127,7 +4127,7 @@ pref("dom.postMessage.sharedArrayBuffer.bypassCOOP_COEP.insecure.enabled", false
 // Whether sites require the open-protocol-handler permission to open a
 //preferred external application for a protocol. If a site doesn't have
 // permission we will show a prompt.
-pref("security.external_protocol_requires_permission", true);
+pref("security.external_protocol_requires_permission", false);
 
 // Preferences for the form autofill toolkit component.
 // The truthy values of "extensions.formautofill.addresses.available"
diff --git a/modules/libpref/moz.build b/modules/libpref/moz.build
index e8f8b97170d3..a1175f7d0528 100644
--- a/modules/libpref/moz.build
+++ b/modules/libpref/moz.build
@@ -45,6 +45,7 @@ pref_groups = [
     "docshell",
     "dom",
     "editor",
+    "embedlite",
     "extensions",
     "fission",
     "font",
diff --git a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
index cf9a2444a3ab..e980e0e2e604 100644
--- a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
+++ b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
@@ -446,5 +446,5 @@ XPCOMUtils.defineLazyPreferenceGetter(
   nsContentDispatchChooser,
   "isPermissionEnabled",
   "security.external_protocol_requires_permission",
-  true
+  false
 );
-- 
2.51.0
